src/object-store/src/layers/lru_cache/read_cache.rs

```diff
@@ -142,18 +142,20 @@ impl<C: Access> ReadCache<C> {
     /// Recover existing cache items from `file_cache` to `mem_cache`.
     /// Return entry count and total approximate entry size in bytes.
     pub(crate) async fn recover_cache(&self) -> Result<(u64, u64)> {
-        let (_, mut pager) = self.file_cache.list("/", OpList::default()).await?;
-
-        while let Some(entry) = pager.next().await? {
+        let op = OperatorBuilder::new(self.file_cache.clone()).finish();
+        let mut entries = op
+            .list_with("/")
+            .metakey(Metakey::ContentLength | Metakey::ContentType)
+            .concurrent(1)
```

Comment:
nit: let's use the const, and how about setting the `concurrent` to 8 or larger?